UI/WPF/Source/Controls/CodeEditorView.xaml.cs

```diff
@@ -1,370 +1,396 @@
+                descriptor.AddValueChanged(CodeLinesScrollViewer, OnViewportHeightChanged);
+
+            descriptor = DependencyPropertyDescriptor.FromProperty(ScrollViewer.VerticalOffsetProperty, typeof(ScrollViewer));
+            if (descriptor != null)
+                descriptor.AddValueChanged(CodeLinesScrollViewer, OnVerticalOffsetChanged);
+
+            descriptor = DependencyPropertyDescriptor.FromProperty(ScrollViewer.HorizontalOffsetProperty, typeof(ScrollViewer));
+            if (descriptor != null)
+                descriptor.AddValueChanged(CodeLinesScrollViewer, OnHorizontalOffsetChanged);
+
+            OnViewportHeightChanged(CodeLinesScrollViewer, EventArgs.Empty);
+            RestoreScrollOffset();
+            EnsureCursorVisible();
+
+            if (ViewModel != null && ViewModel.IsCenterOnLineRequested)
+                CenterOnLine();
+        }
+
+        private ScrollViewer CodeLinesScrollViewer
+        {
+            get { return (ScrollViewer)VisualTreeHelper.GetChild(codeEditorLines, 0); }
+        }
+
+
+        private void RestoreScrollOffset()
+        {
+            if (IsLoaded && ViewModel != null)
+            {
+                double scrollOffset = (double)ViewModel.GetValue(EditorScrollOffsetProperty);
+                double scrollOffsetHorizontal = (double)ViewModel.GetValue(HorizontalScrollOffsetProperty);
+                CodeLinesScrollViewer.ScrollToVerticalOffset(scrollOffset);
+                CodeLinesScrollViewer.ScrollToHorizontalOffset(scrollOffsetHorizontal);
+            }
+        }
+
+        private void OnViewportHeightChanged(object sender, EventArgs e)
+        {
+            if (ViewModel != null)
+                ViewModel.VisibleLines = (int)CodeLinesScrollViewer.ViewportHeight;
+        }
+
+        private static readonly ModelProperty EditorScrollOffsetProperty = ModelProperty.Register(typeof(CodeEditorView), null, typeof(double), 0.0);
+        private static readonly ModelProperty HorizontalScrollOffsetProperty = ModelProperty.Register(typeof(CodeEditorView), null, typeof(double), 0.0);
+
+        private void OnVerticalOffsetChanged(object sender, EventArgs e)
+        {
+            if (ViewModel != null)
+                ViewModel.SetValueCore(EditorScrollOffsetProperty, ((ScrollViewer)sender).VerticalOffset);
+        }
+
+        private void OnHorizontalOffsetChanged(object sender, EventArgs e)
+        {
+            if (ViewModel != null)
+                ViewModel.SetValueCore(HorizontalScrollOffsetProperty, ((ScrollViewer)sender).HorizontalOffset);
+
+            // show / hide the line number border
+            Border linenum = this.FindName("LineNum") as Border;
+            if ( (double)ViewModel.GetValue(HorizontalScrollOffsetProperty) <= 0.1 )
+            {
+                linenum.Visibility = Visibility.Visible;
+            }
+            else
+            {
+                linenum.Visibility = Visibility.Hidden;
+            }

```

Comment:
The line disappears as soon as I start scrolling horizontally, but the line numbers are still visible:

![image](https://user-images.githubusercontent.com/32680403/207754783-9925a221-2a61-4a01-805e-cd6378ce8a32.png)
![image](https://user-images.githubusercontent.com/32680403/207754827-70f52ee3-8427-44fe-957d-b2389c1c8540.png)

Also note that the line appears to go through the scrollbar

Ideally, the line numbers and separator would remain visible and only the content would scroll horizontally.